patches/common/chromium/refactor_pass_the_browsercontext_through_to_getuseragent.patch

```diff
@@ -0,0 +1,274 @@
+ 
+   // Pass bits of global renderer state to the renderer.
+   GetRendererInterface()->SetUserAgent(
+-      GetContentClient()->browser()->GetUserAgent());
++      GetContentClient()->browser()->GetUserAgent(GetBrowserContext()));
+   GetRendererInterface()->SetUserAgentMetadata(
+       GetContentClient()->browser()->GetUserAgentMetadata());
+   NotifyRendererIfLockedToSite();
+diff --git a/content/browser/service_worker/service_worker_process_manager.cc b/content/browser/service_worker/service_worker_process_manager.cc
+index 68718d4f55e89cb5004091f7cfaff9f0df259348..0420fda8e11b9a18daf956fdd25a8e7be38c62aa 100644
+--- a/content/browser/service_worker/service_worker_process_manager.cc
++++ b/content/browser/service_worker/service_worker_process_manager.cc
+@@ -50,6 +50,11 @@ BrowserContext* ServiceWorkerProcessManager::browser_context() {
+   return browser_context_;
+ }
+ 
++std::string ServiceWorkerProcessManager::GetUserAgent(ContentBrowserClient* client) {
++  base::AutoLock lock(browser_context_lock_);
++  return client->GetUserAgent(browser_context_);
++}
++
+ void ServiceWorkerProcessManager::Shutdown() {
+   DCHECK_CURRENTLY_ON(BrowserThread::UI);
+   {
+diff --git a/content/browser/service_worker/service_worker_process_manager.h b/content/browser/service_worker/service_worker_process_manager.h
+index 7eada9bf2d9fd6042a488ee2df7be565c287bcba..9b8d293f2ab61d0c1b22c15f0eb16e7c4afdaf69 100644
+--- a/content/browser/service_worker/service_worker_process_manager.h
++++ b/content/browser/service_worker/service_worker_process_manager.h
+@@ -13,6 +13,7 @@
+ #include "base/gtest_prod_util.h"
+ #include "base/memory/weak_ptr.h"
+ #include "base/synchronization/lock.h"
++#include "content/public/browser/content_browser_client.h"
+ #include "content/browser/service_worker/service_worker_metrics.h"
+ #include "third_party/blink/public/common/service_worker/service_worker_status_code.h"
+ 
+@@ -51,6 +52,8 @@ class CONTENT_EXPORT ServiceWorkerProcessManager {
+   // Called on the UI thread.
+   BrowserContext* browser_context();
+ 
++  std::string GetUserAgent(ContentBrowserClient* client);
++
+   // Synchronously prevents new processes from being allocated
+   // and drops references to RenderProcessHosts. Called on the UI thread.
+   void Shutdown();
+diff --git a/content/browser/service_worker/service_worker_version.cc b/content/browser/service_worker/service_worker_version.cc
+index c6d61cc8795e1eb12f19b5eafec7e80446403afa..0bec70805949f37581eaf8a5e378b557c9aa071d 100644
+--- a/content/browser/service_worker/service_worker_version.cc
++++ b/content/browser/service_worker/service_worker_version.cc
+@@ -1634,7 +1634,7 @@ void ServiceWorkerVersion::StartWorkerInternal() {
+   params->scope = scope_;
+   params->script_url = script_url_;
+   params->script_type = script_type_;
+-  params->user_agent = GetContentClient()->browser()->GetUserAgent();
++  params->user_agent = context_->process_manager()->GetUserAgent(GetContentClient()->browser());
```

Comment:
`GetContentClient()->browser()` can be accessed directly in `ServiceWorkerProcessManager`, since its a per process variable. Pass it as an argument isn't the usual style :)

Reply:
@deepak1556 That's what I thought but the thing failed to build when I tried to use it directly.  I'll try again when I try to upstream